agent/acs/handler/payload_handler.go

```diff
@@ -252,29 +269,46 @@ func (payloadHandler *payloadRequestHandler) addTasks(payload *ecsacs.PayloadMes
 		}
 
 		// Generate an ack request for the credentials in the task, if the
-		// task is associated with an IAM Role
-		taskCredentialsId := task.GetCredentialsID()
-		if taskCredentialsId == "" {
-			// CredentialsId not set for task, no need to ack.
-			continue
+		// task is associated with an IAM role or the exectuion role
+		taskCredentialsID := task.GetCredentialsID()
+		ack, err := payloadHandler.ackCredentials(payload.MessageId, taskCredentialsID)
+		if err != nil {
+			allTasksOK = false
+			seelog.Errorf("ACK task role credentials failed for task: %s, err: %v", task.Arn, err)
+		}
+		if ack != nil {
+			credentialsAcks = append(credentialsAcks, ack)
 		}
 
-		creds, ok := payloadHandler.credentialsManager.GetTaskCredentials(taskCredentialsId)
-		if !ok {
-			seelog.Errorf("Credentials could not be retrieved for task: %s", task.Arn)
+		taskExecutionCredentialsID := task.GetExecutionCredentialsID()
+		ack, err = payloadHandler.ackCredentials(payload.MessageId, taskExecutionCredentialsID)
+		if err != nil {
 			allTasksOK = false
-		} else {
-			credentialsAcks = append(credentialsAcks, &ecsacs.IAMRoleCredentialsAckRequest{
-				MessageId:     payload.MessageId,
-				Expiration:    aws.String(creds.IAMRoleCredentials.Expiration),
-				CredentialsId: aws.String(creds.IAMRoleCredentials.CredentialsID),
-			})
+			seelog.Errorf("ACK task execution role credentials failed for task: %s, err: %v", task.Arn, err)
+		}
+		if ack != nil {
+			credentialsAcks = append(credentialsAcks, ack)
 		}
-
 	}
 	return credentialsAcks, allTasksOK
 }
 
+func (payloadHandler *payloadRequestHandler) ackCredentials(messageID *string, credentialsID string) (*ecsacs.IAMRoleCredentialsAckRequest, error) {
+	if credentialsID == "" {
```

Comment:
Can you refactor this a bit so as to never return `nil, nil`? Returning `nil, nil` makes it so that we are processing 3 states instead of 2 when this function returns. Let's avoid that as much as possible for the sake of simplicity. It should be simple enough to get rid of this and do the following instead in `addTasks`:
```
taskCredentialsID := task.GetCredentialsID()
if taskCredentialsID != "" {
 ack, err := ...
}

...
taskExecutionCredentialsID := task.GetExecutionCredentialsID()
if taskExecutionCredentialsID != "" {
 ack, err  := ...
}
```